### PR TITLE
Remove version gates 

### LINKED ISSUE #3886: Remove version gates from rustfmt codebase
- Remove version gates from rustfmt codebase.
- Add `Three`, `Stable` and `Latest` to `Version`, and set its default value to `Two`.
    - We want to add `Stable` and `Latest` so that users don't need to explicitly change their rustfmt.toml on every major version update.
- Fix and remove tests.

### PR DESCRIPTION
**Pull request by topecongiro**
Close #3886.

### CODE REVIEW

File: src/missed_spans.rs [hunk 1]
```diff
@@ -260,34 +260,9 @@ impl<'a> FmtVisitor<'a> {
             self.config.max_width() - self.block_indent.width(),
         );
         let comment_shape = Shape::legacy(comment_width, comment_indent);
-
-        if on_same_line {
```
Comment (scampi) [inline src/missed_spans.rs]: the `on_same_line` variable is used by the version Two code removed above, and so this block should be re-inserted.

File: src/utils.rs [hunk 2]
```diff
@@ -193,13 +193,13 @@ pub(crate) fn is_attributes_extendable(attrs_str: &str) -> bool {
     !attrs_str.contains('\n') && !last_line_contains_single_line_comment(attrs_str)
 }
 
-/// The width of the first line in s.
+// The width of the first line in s.
```
Comment (scampi) [inline src/utils.rs]: add back the 3rd slash ?

File: src/utils.rs [hunk 3]
```diff
@@ -193,13 +193,13 @@ pub(crate) fn is_attributes_extendable(attrs_str: &str) -> bool {
...
 #[inline]
 pub(crate) fn first_line_width(s: &str) -> usize {
     unicode_str_width(s.splitn(2, '\n').next().unwrap_or(""))
 }
 
-/// The width of the last line in s.
+// The width of the last line in s.
```
Comment (scampi) [inline src/utils.rs]: add back the 3rd slash ?

File: src/visitor.rs [hunk 4]
```diff
@@ -236,103 +236,96 @@ impl<'b, 'a: 'b> FmtVisitor<'a> {
...
-        let mut unindented = false;
-        let mut prev_ends_with_newline = false;
-        let mut extra_newline = false;
+        let mut prev_kind = CodeCharKind::Normal;
+        let mut newline_inserted = false;
 
         let skip_normal = |s: &str| {
             let trimmed = s.trim();
-            trimmed.is_empty() || trimmed.chars().all(|c| c == ';')
+            !trimmed.is_empty() && trimmed.chars().all(|c| c == ';')
```
Comment (scampi) [inline src/visitor.rs]: why is this condition changed ?

File: tests/target/trailing_comments/hard_tabs.rs [hunk 5]
```diff
@@ -12,7 +12,8 @@ impl Foo {
 }
 
 fn lorem_ipsum() {
-	let f = bar(); // Donec consequat mi. Quisque vitae dolor. Integer lobortis. Maecenas id nulla. Lorem.
+	let f = bar(); // Donec consequat mi. Quisque vitae dolor. Integer lobortis. Maecenas id nulla.
+			   // Lorem.
```
Comment (scampi) [inline tests/target/trailing_comments/hard_tabs.rs]: the alignment seems off.

**Comment by topecongiro** [comment]
@scampi Sorry about the PR being large and unorganized and hard to review. I first started by simply removing all the version gates, but it didn't work out. I needed to revert #3833 and implement aligning comments in `CommentCodeSlices`.

File: tests/target/trailing_comments/hard_tabs.rs (hunk 5 above)
Comment (topecongiro) [reply tests/target/trailing_comments/hard_tabs.rs]: Yeah, I think we need to convert `\t` to spaces when calculating the offset.

**Comment by scampi** [comment]
No problem @topecongiro, I thought these commits were included in this PR by mistake, which happened to me more than once ;o)

File: tests/target/trailing_comments/hard_tabs.rs (hunk 5 above)
Comment (topecongiro) [reply tests/target/trailing_comments/hard_tabs.rs]: Filed https://github.com/rust-lang/rustfmt/issues/3950.

**Pull request merged**